rocketpy/plots/rocket_plots.py

```diff
@@ -134,10 +134,22 @@ def drag_curves(self):
         None
         """
 
-        x_power_drag_off = self.rocket.power_off_drag.x_array
-        y_power_drag_off = self.rocket.power_off_drag.y_array
-        x_power_drag_on = self.rocket.power_on_drag.x_array
-        y_power_drag_on = self.rocket.power_on_drag.y_array
+        try:
+            x_power_drag_on = self.rocket.power_on_drag.x_array
+            y_power_drag_on = self.rocket.power_on_drag.y_array
+        except AttributeError:
+            x_power_drag_on = np.linspace(0, 2, 50)
+            y_power_drag_on = np.array(
+                [self.rocket.power_on_drag.source(x) for x in x_power_drag_on]
+            )
+        try:
+            x_power_drag_off = self.rocket.power_off_drag.x_array
+            y_power_drag_off = self.rocket.power_off_drag.y_array
+        except AttributeError:
+            x_power_drag_off = np.linspace(0, 2, 50)
+            y_power_drag_off = np.array(
+                [self.rocket.power_off_drag.source(x) for x in x_power_drag_off]
+            )
```

Comment:
I believe it would be simpler to check whether the `source` is a `callable` and use `set_discrete(mutate_self=False)`.

Another option, perhaps a better use of the `Function` inner plotting mechanisms, would be the `plot(return_object=True)`.

Reply:
I'm not in favor of using the set_discrete method due to 2 reasons:

1 - We can't (or shouldn't) modify the source just to plot this data
2 - Even if we don't mutate the source, we would still take longer times to discretize the Function rather then just evaluating it in a few data points.

I understand the importance of reusing the code and features provided from Function. But im this case I prefer the fastest, easier solution, which is to simply add this try/except in the code.